monitored-httpclient/src/main/java/com/github/nhenneaux/resilienthttpclient/monitoredclientpool/RoundRobinPool.java

```diff
@@ -24,7 +25,7 @@ class RoundRobinPool {
      *
      * @param list a list of service instance.
      */
-    RoundRobinPool(final List<SingleIpHttpClient> list) {
+    RoundRobinPool(final Collection<SingleIpHttpClient> list) {
```

Comment:
Not always a list?

Reply:
Nope. Using `Set` to improve checking with `::contains`

Reply:
ordering is important here so I don't think a Set is harmless